src/vulkan/pipeline.cc

```diff
@@ -159,36 +179,57 @@ Result Pipeline::CreateDescriptorPools() {
         VK_SUCCESS) {
       return Result("Vulkan::Calling vkCreateDescriptorPool Fail");
     }
-    descriptor_pools_.push_back(desc_pool);
+
+    if (static_cast<uint32_t>(descriptor_sets_.size()) <= current_desc) {
+      return Result(
+          "Pipeline::CreateDescriptorPools more actually used descriptor sets "
+          "than descriptor set layouts");
```

Comment:
"Pipeline::CreateDescriptorPools: More descriptor sets requested then descriptor set layouts provided"

Reply:
done
I used "than" not "then"